### PR TITLE
EVA-2669 - Script to copy blocks from dev allocation to prod

### PR DESCRIPTION
**Pull request by tcezard**

### CODE REVIEW

File: tasks/eva_2669/copy_allocation_blocks.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+             f"and first_value >= 3166700000;")
+    rows = get_all_results_for_query(dev_conn, query)
+
+    # Write them out in prod
+    for row in rows:
+        query = 'INSERT INTO public.contiguous_id_blocks VALUES ' + serialise_tuple(row)
+        print(query)
+        if args.apply_queries:
+            prod_cursor.execute(query)
+
```
Comment (tcezard) [inline tasks/eva_2669/copy_allocation_blocks.py:49]: Should I commit the first set of queries before I get the hibernate_sequence value?

File: tasks/eva_2669/copy_allocation_blocks.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    query = (f'select application_instance_id, category_id, "first_value", last_committed , "last_value" '
+             f'from public.contiguous_id_blocks '
+             f"where category_id ='rs' "
+             f"and application_instance_id = 'instance-6'"
+             f"and first_value >= 3166700000;")
+    rows = get_all_results_for_query(dev_conn, query)
+
+    # Write them out in prod
+    for row in rows:
+        query = 'INSERT INTO public.contiguous_id_blocks VALUES ' + serialise_tuple(row)
```
Comment (sundarvenkata-EBI) [inline tasks/eva_2669/copy_allocation_blocks.py:57]: I don't see ID being inserted here.

Comment (tcezard) [reply tasks/eva_2669/copy_allocation_blocks.py:57]: I though this was an auto increment. and I should let the db do it. Did I get that wrong ?

Comment (tcezard) [reply tasks/eva_2669/copy_allocation_blocks.py]: I've do the increment in the script now.

File: tasks/eva_2669/copy_allocation_blocks.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    # Get the newest existing block from prod
+    query = """select max("first_value") from public.contiguous_id_blocks where category_id ='rs' """
+    largest_first_value_in_prod, = get_all_results_for_query(prod_conn, query)[0]
+
+    # Fill in with incomplete blocks until reaching the first block used in dev
+    for first_value in range(largest_first_value_in_prod + 100000, 3166700000, 100000):
+        last_id_used += 1
+        query = 'INSERT INTO public.contiguous_id_blocks VALUES ' + \
+                f"({last_id_used}, 'instance-6', 'rs', {first_value}, {first_value}, {first_value + 999999})"
```
Comment (sundarvenkata-EBI) [inline tasks/eva_2669/copy_allocation_blocks.py:36]:
```suggestion
                f"({last_id_used}, 'instance-6', 'rs', {first_value}, {first_value}, {first_value + 99999})"
```

File: tasks/eva_2669/copy_allocation_blocks.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+             f"and application_instance_id = 'instance-6'"
+             f"and first_value >= 3166700000;")
+    rows = get_all_results_for_query(dev_conn, query)
+
+    # Write them out in prod
+    for application_instance_id, category_id, first_value, last_committed , last_value in rows:
+        last_id_used += 1
+        query = 'INSERT INTO public.contiguous_id_blocks VALUES ' + \
+                f"({last_id_used}, '{application_instance_id}', '{category_id}', {first_value}, {last_committed}, " \
+                f"{last_value})"
```
[the remark this answers is not in the document]
Comment (tcezard) [reply tasks/eva_2669/copy_allocation_blocks.py:54]: If there is any failure the cursor would not commit so it wont have created anything.
With the `not exist` it becomes difficult to know if the entry has been added and keep track of the `last_id_used`.

[the remark this answers is not in the document]
Comment (sundarvenkata-EBI) [reply tasks/eva_2669/copy_allocation_blocks.py:54]: Makes sense.

**Review by sundarvenkata-EBI — APPROVED** [review]

File: tasks/eva_2669/copy_allocation_blocks.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+    # Get the newest existing block from prod
+    query = """select max("first_value") from public.contiguous_id_blocks where category_id ='rs' """
+    largest_first_value_in_prod, = get_all_results_for_query(prod_conn, query)[0]
+
+    # Fill in with incomplete blocks until reaching the first block used in dev
+    for first_value in range(largest_first_value_in_prod + 100000, 3166700000, 100000):
+        last_id_used += 1
+        query = 'INSERT INTO public.contiguous_id_blocks VALUES ' + \
+                f"({last_id_used}, 'instance-6', 'rs', {first_value}, {first_value}, {first_value + 99999})"
```
Comment (andresfsilva) [inline tasks/eva_2669/copy_allocation_blocks.py:36]:
```suggestion
                f"({last_id_used}, 'instance-6', 'rs', {first_value}, {first_value - 1}, {first_value + 99999})"
```

**Review by andresfsilva — CHANGES_REQUESTED** [review]

**Review by andresfsilva — APPROVED** [review]

**Pull request merged**